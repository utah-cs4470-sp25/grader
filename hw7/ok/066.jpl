struct mpair {
  a: int
  b: int
}

fn f(mp: mpair) : int {
   return mp.a - mp.b
}
show f(mpair{1, 2})
