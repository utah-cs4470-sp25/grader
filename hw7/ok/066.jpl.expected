(StructCmd mpair a (IntType) b (IntType))
(FnCmd f (((VarLValue mp) (StructType mpair))) (IntType) (ReturnStmt (BinopExpr (IntType) (DotExpr (IntType) (VarExpr (StructType mpair) mp) a) - (DotExpr (IntType) (VarExpr (StructType mpair) mp) b))))
(ShowCmd (CallExpr (IntType) f (StructLiteralExpr (StructType mpair) mpair (IntExpr (IntType) 1) (IntExpr (IntType) 2))))
